### PR TITLE
Schema registry update kafka zk

### PR DESCRIPTION
**Pull request by benjigoldberg**
Updating Schema Registry Kafka dependency to 0.8.5 to keep up with the latest. This required a few minor changes to the subchart configuration:
* default persistence off
* change the zookeeper `servers` variable to the new `replicaCount` variable

Also added some missing documentation to README and the values file.

### CODE REVIEW

**Comment by benjigoldberg** [comment]
@unguiculus @viglesiasce if either of you have a moment to take a quick glance at these changes it would be much appreciated!

File: incubator/schema-registry/values.yaml [hunk 1]
```diff
@@ -90,20 +90,31 @@ sasl:
...
+  ## Override kafka settings for default installations
   configurationOverrides:
     # Needed to run with 1 Kafka Broker
     offsets.topic.replication.factor: 1
+  ## Run only a single kafka broker by default
   replicas: 1
+  ## Do not enable kafka persistence by default. This setting is only acceptable in test
+  ## environments.
+  persistence:
+    enabled: false
```
Comment (davidkarlsen) [inline incubator/schema-registry/values.yaml]: is a more sane value to enable persistence by default? see https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#Persistence

File: incubator/schema-registry/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: schema-registry
 home: https://docs.confluent.io/current/schema-registry/docs/index.html
-version: 0.4.5
-appVersion: 4.0.1
+version: 0.4.6
```
Comment (davidkarlsen) [inline incubator/schema-registry/Chart.yaml]: since some parameters are removed/change name, would it make more sense to change the major version?

**Comment by benjigoldberg** [comment]
@davidkarlsen thank you for the review! Excellent points -- addressed both

File: incubator/schema-registry/README.md [hunk 3]
```diff
@@ -82,10 +82,13 @@ The following table lists the configurable parameters of the SchemaRegistry char
 | `kafkaStore.overrideBootstrapServers` | Defaults to Kafka Servers in the same release, it can be overridden in case there was a separate release for Kafka Deploy | `{{- printf "PLAINTEXT://%s-kafka-headless:9092" .Release.Name }}`
 | `kafka.enabled` | If `true`, install Kafka/Zookeeper alongside the `SchemaRegistry`. This is intended for testing and argument-less helm installs of this chart only and should not be used in Production. | `true` |
 | `kafka.replicas` | The number of Kafka Pods to install as part of the `StatefulSet` if `kafka.Enabled` is `true`| `1` |
+| `kafka.configurationOverrides` | Any Kafka Configuration overrides to provide to the underlying kafka chart | `{offsets.topic.replica.factor: 1}` |
+| `kafka.persistence.enabled` | Whether or not to use a persistent volume for kafka | `false` |
```
Comment (davidkarlsen) [inline incubator/schema-registry/README.md]: this does not match the default

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

**Comment by benjigoldberg** [comment]
@davidkarlsen updated again -- thank you for the detailed review 👍

**Pull request merged**